app/components/inputs/InputComponent.js

```diff
@@ -0,0 +1,188 @@
+    onFocus && onFocus(e);
+  };
+  onInputBlur = (e) => {
+    const { onBlur } = this.props;
+    const { inputUnitDiv } = this.state;
+    var updatedInputUnitDiv = inputUnitDiv;
+    this.setState({ inputUnitDiv: updatedInputUnitDiv.classList.remove("active") });
+    onBlur && onBlur(e);
+  };
+  onKeyDown = (e) => {
+    const { onKeyDownSubmit, onKeyDown } = this.props;
+    (e.keyCode === 13) && onKeyDownSubmit && onKeyDownSubmit(e);
+    onKeyDown && !e.defaultPrevented && onKeyDown(e);
+  }
+  onKeyDown2 = (e) => {
+    if (e.keyCode === 13) {
+      e.preventDefault();
+    }
+    const { onKeyDownSubmit, onKeyDown } = this.props;
+    (e.keyCode === 13) && onKeyDownSubmit && onKeyDownSubmit(e);
+    onKeyDown && !e.defaultPrevented && onKeyDown(e);
+  }
+  onChange = () => {
+    this.setState({ inputPhase: 1 });
+    var number = this.state.inputMain + this.state.inputSub;
+    this.setState({ inputVal: number });
+  };
+  onChange2 = (e) => {
+    this.setState({ inputPhase: 0 });
+    var number = e.target.textContent;
+    var number_f = parseFloat(number);
+    var left_part = number_f.toFixed(2);
+    var right_part_raw = number_f - left_part;
+    var right_part_rounded = parseFloat(String(right_part_raw.toFixed(10)).slice(0, -1));
+
+    var right_part = right_part_rounded * 100000000;
+
+    // Used to compare value later
+    var prepad = right_part_rounded * 100000000;
+
+    right_part = right_part.toFixed(0);
+
+    // Deduce rounding errors
+    if (Math.floor(prepad) < right_part && Math.floor(prepad) == (parseInt(right_part)) - 1 ) {
+      right_part = (parseInt(right_part)) - 1;
+    }
+
+    // Deduce rounding errors (upto 10 decimal places)
+    else if (Math.floor(right_part_raw * 100000000) < right_part && Math.floor(right_part_raw * 100000000) == (parseInt(right_part)) - 1) {
+      right_part = (parseInt(right_part)) - 1;
+    }
+
+    // Pad for numbers, so 0032 decimal will not turn to 32
+    right_part = right_part.toString().padStart(6, "0");
+
+    if (right_part <= 0) {
+      right_part = "000000";
+    }
+
+    // Reset back to numbers
+    if (isNaN(left_part) == true || isNaN(right_part) == true) {
+      left_part = "0.00";
+      right_part = "000000";
+    }
+
+    this.setState({ inputMain: left_part });
+    this.setState({ inputSub: right_part });
+
+    // Send to other older previous unstyled component and trigger event
+    document.getElementById("parsedInput").value = number_f;
+    var event = new Event("input", { bubbles: true });
+    var x = document.getElementById("parsedInput");
+    var tracker = x._valueTracker;
+    tracker.setValue("'" + x.value + "'");
+    x.dispatchEvent(event);
+  };
+
+  render() {
+    const {
+      showErrors,
+      invalidMessage,
+      requiredMessage,
+      required,
+      invalid,
+      value,
+      placeholder,
+      onChange,
+      disabled,
+      readOnly,
+      unit,
+      hidden,
+      type,
+    } = this.props;
```

Comment:
coming from `this.props` (ie, the props injected into this component by whoever is using it).